feat: implements https://github.com/wtfnotavailable/Discord-MusicBot/…

brianferri opened the pull request:
## Please describe the changes this PR makes and why it should be merged:
Implements https://github.com/wtfnotavailable/Discord-MusicBot/issues/71

<!--
Please move lines that apply to you out of the comment:
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)
- This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- This PR **only** includes non-code changes, like changes to documentation, README, etc.
-->

# Important.

- Write in camelCase, not snake_case.
- Do not push to master/main without testing your changes first, make a branch
  if you have to.

brianferri committed:
feat: implements https://github.com/wtfnotavailable/Discord-MusicBot/issues/71
chore: updates packages (revert this file if incompatibilities arise)

Neko-Life committed:
WIP: refactor this to comply to DRY principle. Someone else can continue or me later on lunch break

Neko-Life commented on djs-bot/commands/music/playlists.js:
```diff
@@ -196,72 +349,252 @@ module.exports = new SlashCommand()
+				],
+				ephemeral: true,
+			});
+		} else if (operation === "play") {
```
Refactor this horrendous if else and copy pasted abomination to use DRY principle. Anyone can continue or me later on lunch break.

dependabot[bot] committed:
chore(deps-dev): bump @types/node from 20.5.9 to 20.6.0 in /dashboard

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 20.5.9 to 20.6.0.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps-dev): bump eslint from 8.48.0 to 8.49.0 in /dashboard

Bumps [eslint](https://github.com/eslint/eslint) from 8.48.0 to 8.49.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v8.48.0...v8.49.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Neko-Life committed:
untested refactor, TODO: testing

Neko-Life commented:
Untested, any issue found can be fixed later

brianferri committed:
chore(deps-dev): bump eslint from 8.48.0 to 8.49.0 in /dashboard (#90)

Bumps [eslint](https://github.com/eslint/eslint) from 8.48.0 to 8.49.0.
<details>
<summary>Release notes</summary>
<p><em>Sourced from <a
href="https://github.com/eslint/eslint/releases">eslint's
releases</a>.</em></p>
<blockquote>
<h2>v8.49.0</h2>
<h2>Features</h2>
<ul>
<li><a
href="https://github.com/eslint/eslint/commit/da09f4e641141f585ef611c6e9d63d4331054706"><code>da09f4e</code></a>
feat: Implement onUnreachableCodePathStart/End (<a
href="https://redirect.github.com/eslint/eslint/issues/17511">#17511</a>)
(Nicholas C. Zakas)</li>
<li><a
href="https://github.com/eslint/eslint/commit/32b2327aafdd3b911fabab69ed75c9ff97658c60"><code>32b2327</code></a>
feat: Emit deprecation warnings in RuleTester (<a
href="https://redirect.github.com/eslint/eslint/issues/17527">#17527</a>)
(Nicholas C. Zakas)</li>
<li><a
href="https://github.com/eslint/eslint/commit/acb7df35b9a7485f26bc6b3e1f9083d1c585dce9"><code>acb7df3</code></a>
feat: add new <code>enforce</code> option to
<code>lines-between-class-members</code> (<a
href="https://redirect.github.com/eslint/eslint/issues/17462">#17462</a>)
(Nitin Kumar)</li>
</ul>
<h2>Documentation</h2>
<ul>
<li><a
href="https://github.com/eslint/eslint/commit/ecfb54ff4cdd18f28b4f9b78f0a78fb4cf80f1b8"><code>ecfb54f</code></a>
docs: Update README (GitHub Actions Bot)</li>
<li><a
href="https://github.com/eslint/eslint/commit/de86b3b2e58edd5826200c23255d8325abe375e1"><code>de86b3b</code></a>
docs: update <code>no-promise-executor-return</code> examples (<a
href="https://redirect.github.com/eslint/eslint/issues/17529">#17529</a>)
(Nitin Kumar)</li>
<li><a
href="https://github.com/eslint/eslint/commit/032c4b1476a7b8cfd917a66772d2221950ea87eb"><code>032c4b1</code></a>
docs: add typescript template (<a
href="https://redirect.github.com/eslint/eslint/issues/17500">#17500</a>)
(James)</li>
<li><a
href="https://github.com/eslint/eslint/commit/cd7da5cc3154f86f7ca45fb58929d27a7af359ed"><code>cd7da5c</code></a>
docs: Update README (GitHub Actions Bot)</li>
</ul>
<h2>Chores</h2>
<ul>
<li><a
href="https://github.com/eslint/eslint/commit/b7621c3b16cf7d5539f05336a827e1b32d95e6ac"><code>b7621c3</code></a>
chore: remove browser test from <code>npm test</code> (<a
href="https://redirect.github.com/eslint/eslint/issues/17550">#17550</a>)
(Milos Djermanovic)</li>
<li><a
href="https://github.com/eslint/eslint/commit/cac45d04b890b0700dd8908927300608adad05fe"><code>cac45d0</code></a>
chore: upgrade <code>@​eslint/js</code><a
href="https://github.com/8"><code>@​8</code></a>.49.0 (<a
href="https://redirect.github.com/eslint/eslint/issues/17549">#17549</a>)
(Milos Djermanovic)</li>
<li><a
href="https://github.com/eslint/eslint/commit/cd395082bffcb4b68efa09226d7c682cef56179e"><code>cd39508</code></a>
chore: package.json update for <code>@​eslint/js</code> release (ESLint
Jenkins)</li>
<li><a
href="https://github.com/eslint/eslint/commit/203a971c0abc3a95ae02ff74104a01e569707060"><code>203a971</code></a>
ci: bump actions/checkout from 3 to 4 (<a
href="https://redirect.github.com/eslint/eslint/issues/17530">#17530</a>)
(dependabot[bot])</li>
<li><a
href="https://github.com/eslint/eslint/commit/a40fa509922b36bb986eb1be9394591f84f62d9e"><code>a40fa50</code></a>
chore: use eslint-plugin-jsdoc's flat config (<a
href="https://redirect.github.com/eslint/eslint/issues/17516">#17516</a>)
(Milos Djermanovic)</li>
<li><a
href="https://github.com/eslint/eslint/commit/926a28684282aeec37680bbc52a66973b8055f54"><code>926a286</code></a>
test: replace Karma with Webdriver.IO (<a
href="https://redirect.github.com/eslint/eslint/issues/17126">#17126</a>)
(Christian Bromann)</li>
<li><a
href="https://github.com/eslint/eslint/commit/f591d2c88bf15af72e3a207b34fa872b4b90464b"><code>f591d2c</code></a>
chore: Upgrade config-array (<a
href="https://redirect.github.com/eslint/eslint/issues/17512">#17512</a>)
(Nicholas C. Zakas)</li>
</ul>
</blockquote>
</details>
<details>
<summary>Changelog</summary>
<p><em>Sourced from <a
href="https://github.com/eslint/eslint/blob/main/CHANGELOG.md">eslint's
changelog</a>.</em></p>
<blockquote>
<p>v8.49.0 - September 8, 2023</p>
<ul>
<li><a
href="https://github.com/eslint/eslint/commit/b7621c3b16cf7d5539f05336a827e1b32d95e6ac"><code>b7621c3</code></a>
chore: remove browser test from <code>npm test</code> (<a
href="https://redirect.github.com/eslint/eslint/issues/17550">#17550</a>)
(Milos Djermanovic)</li>
<li><a
href="https://github.com/eslint/eslint/commit/cac45d04b890b0700dd8908927300608adad05fe"><code>cac45d0</code></a>
chore: upgrade <code>@​eslint/js</code><a
href="https://github.com/8"><code>@​8</code></a>.49.0 (<a
href="https://redirect.github.com/eslint/eslint/issues/17549">#17549</a>)
(Milos Djermanovic)</li>
<li><a
href="https://github.com/eslint/eslint/commit/cd395082bffcb4b68efa09226d7c682cef56179e"><code>cd39508</code></a>
chore: package.json update for <code>@​eslint/js</code> release (ESLint
Jenkins)</li>
<li><a
href="https://github.com/eslint/eslint/commit/ecfb54ff4cdd18f28b4f9b78f0a78fb4cf80f1b8"><code>ecfb54f</code></a>
docs: Update README (GitHub Actions Bot)</li>
<li><a
href="https://github.com/eslint/eslint/commit/da09f4e641141f585ef611c6e9d63d4331054706"><code>da09f4e</code></a>
feat: Implement onUnreachableCodePathStart/End (<a
href="https://redirect.github.com/eslint/eslint/issues/17511">#17511</a>)
(Nicholas C. Zakas)</li>
<li><a
href="https://github.com/eslint/eslint/commit/de86b3b2e58edd5826200c23255d8325abe375e1"><code>de86b3b</code></a>
docs: update <code>no-promise-executor-return</code> examples (<a
href="https://redirect.github.com/eslint/eslint/issues/17529">#17529</a>)
(Nitin Kumar)</li>
<li><a
href="https://github.com/eslint/eslint/commit/203a971c0abc3a95ae02ff74104a01e569707060"><code>203a971</code></a>
ci: bump actions/checkout from 3 to 4 (<a
href="https://redirect.github.com/eslint/eslint/issues/17530">#17530</a>)
(dependabot[bot])</li>
<li><a
href="https://github.com/eslint/eslint/commit/32b2327aafdd3b911fabab69ed75c9ff97658c60"><code>32b2327</code></a>
feat: Emit deprecation warnings in RuleTester (<a
href="https://redirect.github.com/eslint/eslint/issues/17527">#17527</a>)
(Nicholas C. Zakas)</li>
<li><a
href="https://github.com/eslint/eslint/commit/acb7df35b9a7485f26bc6b3e1f9083d1c585dce9"><code>acb7df3</code></a>
feat: add new <code>enforce</code> option to
<code>lines-between-class-members</code> (<a
href="https://redirect.github.com/eslint/eslint/issues/17462">#17462</a>)
(Nitin Kumar)</li>
<li><a
href="https://github.com/eslint/eslint/commit/032c4b1476a7b8cfd917a66772d2221950ea87eb"><code>032c4b1</code></a>
docs: add typescript template (<a
href="https://redirect.github.com/eslint/eslint/issues/17500">#17500</a>)
(James)</li>
<li><a
href="https://github.com/eslint/eslint/commit/cd7da5cc3154f86f7ca45fb58929d27a7af359ed"><code>cd7da5c</code></a>
docs: Update README (GitHub Actions Bot)</li>
<li><a
href="https://github.com/eslint/eslint/commit/a40fa509922b36bb986eb1be9394591f84f62d9e"><code>a40fa50</code></a>
chore: use eslint-plugin-jsdoc's flat config (<a
href="https://redirect.github.com/eslint/eslint/issues/17516">#17516</a>)
(Milos Djermanovic)</li>
<li><a
href="https://github.com/eslint/eslint/commit/926a28684282aeec37680bbc52a66973b8055f54"><code>926a286</code></a>
test: replace Karma with Webdriver.IO (<a
href="https://redirect.github.com/eslint/eslint/issues/17126">#17126</a>)
(Christian Bromann)</li>
<li><a
href="https://github.com/eslint/eslint/commit/f591d2c88bf15af72e3a207b34fa872b4b90464b"><code>f591d2c</code></a>
chore: Upgrade config-array (<a
href="https://redirect.github.com/eslint/eslint/issues/17512">#17512</a>)
(Nicholas C. Zakas)</li>
</ul>
</blockquote>
</details>
<details>
<summary>Commits</summary>
<ul>
<li><a
href="https://github.com/eslint/eslint/commit/55c1685aab0fa75074ac1033abfa6e34cb7baaf6"><code>55c1685</code></a>
8.49.0</li>
<li><a
href="https://github.com/eslint/eslint/commit/d498a5197399499a2c0566d5124c3235a8976ef3"><code>d498a51</code></a>
Build: changelog update for 8.49.0</li>
<li><a
href="https://github.com/eslint/eslint/commit/b7621c3b16cf7d5539f05336a827e1b32d95e6ac"><code>b7621c3</code></a>
chore: remove browser test from <code>npm test</code> (<a
href="https://redirect.github.com/eslint/eslint/issues/17550">#17550</a>)</li>
<li><a
href="https://github.com/eslint/eslint/commit/cac45d04b890b0700dd8908927300608adad05fe"><code>cac45d0</code></a>
chore: upgrade <code>@​eslint/js</code><a
href="https://github.com/8"><code>@​8</code></a>.49.0 (<a
href="https://redirect.github.com/eslint/eslint/issues/17549">#17549</a>)</li>
<li><a
href="https://github.com/eslint/eslint/commit/cd395082bffcb4b68efa09226d7c682cef56179e"><code>cd39508</code></a>
chore: package.json update for <code>@​eslint/js</code> release</li>
<li><a
href="https://github.com/eslint/eslint/commit/ecfb54ff4cdd18f28b4f9b78f0a78fb4cf80f1b8"><code>ecfb54f</code></a>
docs: Update README</li>
<li><a
href="https://github.com/eslint/eslint/commit/da09f4e641141f585ef611c6e9d63d4331054706"><code>da09f4e</code></a>
feat: Implement onUnreachableCodePathStart/End (<a
href="https://redirect.github.com/eslint/eslint/issues/17511">#17511</a>)</li>
<li><a
href="https://github.com/eslint/eslint/commit/de86b3b2e58edd5826200c23255d8325abe375e1"><code>de86b3b</code></a>
docs: update <code>no-promise-executor-return</code> examples (<a
href="https://redirect.github.com/eslint/eslint/issues/17529">#17529</a>)</li>
<li><a
href="https://github.com/eslint/eslint/commit/203a971c0abc3a95ae02ff74104a01e569707060"><code>203a971</code></a>
ci: bump actions/checkout from 3 to 4 (<a
href="https://redirect.github.com/eslint/eslint/issues/17530">#17530</a>)</li>
<li><a
href="https://github.com/eslint/eslint/commit/32b2327aafdd3b911fabab69ed75c9ff97658c60"><code>32b2327</code></a>
feat: Emit deprecation warnings in RuleTester (<a
href="https://redirect.github.com/eslint/eslint/issues/17527">#17527</a>)</li>
<li>Additional commits viewable in <a
href="https://github.com/eslint/eslint/compare/v8.48.0...v8.49.0">compare
view</a></li>
</ul>
</details>
<br />


[![Dependabot compatibility
score](https://dependabot-badges.githubapp.com/badges/compatibility_score?dependency-name=eslint&package-manager=npm_and_yarn&previous-version=8.48.0&new-version=8.49.0)](https://docs.github.com/en/github/managing-security-vulnerabilities/about-dependabot-security-updates#about-compatibility-scores)

Dependabot will resolve any conflicts with this PR as long as you don't
alter it yourself. You can also trigger a rebase manually by commenting
`@dependabot rebase`.

[//]: # (dependabot-automerge-start)
[//]: # (dependabot-automerge-end)

---

<details>
<summary>Dependabot commands and options</summary>
<br />

You can trigger Dependabot actions by commenting on this PR:
- `@dependabot rebase` will rebase this PR
- `@dependabot recreate` will recreate this PR, overwriting any edits
that have been made to it
- `@dependabot merge` will merge this PR after your CI passes on it
- `@dependabot squash and merge` will squash and merge this PR after
your CI passes on it
- `@dependabot cancel merge` will cancel a previously requested merge
and block automerging
- `@dependabot reopen` will reopen this PR if it is closed
- `@dependabot close` will close this PR and stop Dependabot recreating
it. You can achieve the same result by closing it manually
- `@dependabot show <dependency name> ignore conditions` will show all
of the ignore conditions of the specified dependency
- `@dependabot ignore this major version` will close this PR and stop
Dependabot creating any more for this major version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this minor version` will close this PR and stop
Dependabot creating any more for this minor version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this dependency` will close this PR and stop
Dependabot creating any more for this dependency (unless you reopen the
PR or upgrade to it yourself)


</details>

brianferri committed:
chore(deps-dev): bump @types/node from 20.5.9 to 20.6.0 in /dashboard (#89)

Bumps
[@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node)
from 20.5.9 to 20.6.0.
<details>
<summary>Commits</summary>
<ul>
<li>See full diff in <a
href="https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node">compare
view</a></li>
</ul>
</details>
<br />


[![Dependabot compatibility
score](https://dependabot-badges.githubapp.com/badges/compatibility_score?dependency-name=@types/node&package-manager=npm_and_yarn&previous-version=20.5.9&new-version=20.6.0)](https://docs.github.com/en/github/managing-security-vulnerabilities/about-dependabot-security-updates#about-compatibility-scores)

Dependabot will resolve any conflicts with this PR as long as you don't
alter it yourself. You can also trigger a rebase manually by commenting
`@dependabot rebase`.

[//]: # (dependabot-automerge-start)
[//]: # (dependabot-automerge-end)

---

<details>
<summary>Dependabot commands and options</summary>
<br />

You can trigger Dependabot actions by commenting on this PR:
- `@dependabot rebase` will rebase this PR
- `@dependabot recreate` will recreate this PR, overwriting any edits
that have been made to it
- `@dependabot merge` will merge this PR after your CI passes on it
- `@dependabot squash and merge` will squash and merge this PR after
your CI passes on it
- `@dependabot cancel merge` will cancel a previously requested merge
and block automerging
- `@dependabot reopen` will reopen this PR if it is closed
- `@dependabot close` will close this PR and stop Dependabot recreating
it. You can achieve the same result by closing it manually
- `@dependabot show <dependency name> ignore conditions` will show all
of the ignore conditions of the specified dependency
- `@dependabot ignore this major version` will close this PR and stop
Dependabot creating any more for this major version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this minor version` will close this PR and stop
Dependabot creating any more for this minor version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this dependency` will close this PR and stop
Dependabot creating any more for this dependency (unless you reopen the
PR or upgrade to it yourself)


</details>
